framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/AClassWrapper.java

```diff
@@ -0,0 +1,191 @@
+        String methodName = JVMNames.getJVMMethodName(methodElt);
+        if (methods.containsKey(methodName)) {
+            return methods.get(methodName);
+        } else {
+            AMethodWrapper wrapper =
+                    new AMethodWrapper(
+                            theClass.methods.getVivify(methodName), methodElt.getReturnType());
+            methods.put(methodName, wrapper);
+            return wrapper;
+        }
+    }
+
+    /**
+     * Get all the methods that have been vivified (that is, interacted with) on a class.
+     *
+     * @return a map from method name to the object representing the method
+     */
+    public Map<String, AMethodWrapper> getMethods() {
+        return ImmutableMap.copyOf(methods);
+    }
+
+    /**
+     * Call before doing anything with a field. Interacts with scenelib and fetches or creates an
+     * appropriate AFieldWrapper object.
+     *
+     * <p>Results are interned.
+     *
+     * @param fieldName the name of the field
+     * @param type the type of the field, which scenelib doesn't track
+     * @return an AField object representing the field
+     */
+    public AFieldWrapper vivifyField(String fieldName, TypeMirror type) {
+        if (fields.containsKey(fieldName)) {
+            return fields.get(fieldName);
+        } else {
+            AFieldWrapper wrapper =
+                    new AFieldWrapper(theClass.fields.getVivify(fieldName), type.toString());
+            fields.put(fieldName, wrapper);
+            return wrapper;
+        }
+    }
+
+    /**
+     * Get all the fields that have been vivified (that is, interacted with) on a class.
+     *
+     * @return a map from field name to the object representing the field
+     */
+    public Map<String, AFieldWrapper> getFields() {
+        return ImmutableMap.copyOf(fields);
+    }
+
+    /**
+     * Get the annotations on the class.
+     *
+     * @return the annotations, directly from scenelib
+     */
+    public Collection<? extends Annotation> getAnnotations() {
+        return theClass.tlAnnotationsHere;
+    }
+
+    /**
+     * Get the type of the class, or null if it is unknown. Callers should ensure that either: -
+     * {@link #setTypeElement(TypeElement)} has been called, or - the return value is checked
+     * against null
+     *
+     * @return a type element representing this class
+     */
+    public @Nullable TypeElement getTypeElement() {
+        return typeElement;
+    }
+
+    /**
+     * Set the type element representing the class.
+     *
+     * @param typeElement the type element representing the class
+     */
+    public void setTypeElement(TypeElement typeElement) {
+        if (this.typeElement == null) {
+            this.typeElement = typeElement;
+        } else if (!this.typeElement.equals(typeElement)) {
+            throw new BugInCF(
+                    "Set the base type of an AClassWrapper to a different type than the previous"
```

Comment:
This sounds like an imperative: the user should set the base type.
A BugInCF error message can be brief, but it should be informative:  give the two types.